dts/bindings/clock/st,stm32f4-pll-clock.yaml

```diff
@@ -0,0 +1,74 @@
+# Copyright (c) 2021, Linaro ltd
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  STM32L4 PLL node.
+
+  It can be used to describe 2 different PLLs: PLL and PLLI2S.
+  Only main PLL is supported for now.
+
+  These PLLs could take one of clk_hse or clk_hsi as input clock, with an
+  input frequency from 1 to 2 MHz. PLLM factor is used to set the input clock
+  in this acceptable range.
+
+  Each PLL can have up to 2 output clocks and for each output clock, the
+  frequency can be computed with the following formulae:
+
+    f(PLL_P) = f(VCO clock) / PLLP  --> PLLCLK (System Clock)
+    f(PLL_Q) = f(VCO clock) / PLLQ  --> PLL48CLK
+
+      with f(VCO clock) = f(PLL clock input) × (PLLN / PLLM)
+
+  The PLL output frequency must not exceed 80 MHz.
+
+
+compatible: "st,stm32f4-pll-clock"
+
+include: [clock-controller.yaml, base.yaml]
+
+properties:
+    "#clock-cells":
+      const: 0
+
+    clocks:
+      required: true
+
+    div-m:
+      type: int
+      required: true
+      description: |
+          Division factor for the PLL input clock
+          Valid range: 2 - 63
+
+    mul-n:
+      type: int
+      required: true
+      description: |
+          Main PLL multiplication factor for VCO
+          Valid range: 50 - 432
+
+    div-p:
+      type: int
+      required: false
```

Comment:
This is required, otherwise the driver will not compile (to verify comment it out in board dts file and run `west build -b nucleo_f429zi samples/basic/blinky -p`). 

Could be the same for `div-p` and `div-q` of other MCU series. I was not sure why just those two divisors were declared as non-required in all bindings. Was this on purpose?

Reply:
There are 2 things to consider:

#### In the cock driver implementation:
The required div depends on implementation:
-F4: Only div-P is required. Q is optional
-L4/WB: Only div-R is required. Q is optional

#### Considering the bindings:
So far (in the series I have worked on), on a series, various PLLs are made out of the same hardware block (ie various factors support the same ranges).
So, in order to limit the number of bindings files to be created, I tried to take advantage of that and used only one compatible to describe all PLL of SoC. For instance, compatible `st,stm32l4-pll-clock` could be used to describe PLL, PLLSAI1 and PLLSAI2 (which is purely theoretical for now as only the main one being implemented in clock driver).

On F4 the same rule applies, but there's a trick: PLL are identical, but don't use the same output:
Main PLL uses output P (-> PLLCLK) and output Q (-> PLL48CLK)
PLLI2S uses output R (PLLI2SCLK)
So, instead of creating 2 compatibles that would be extremely close, I've created one compatible with optional (/not required) outputs.
Alternative would be to create `st,stm32f4-plli2s-clock`, inheriting from `st,stm32f4-pll-clock`, deleting div-p and div-q and  adding div-r, and setting all as `required: true`. This is an interesting option: more "binding".

This whole summary helped to change my mind in favor of this last option, as I think this should help minimizing users errors. Thanks for this question !

Reply:
Thanks for the explanation. Honestly I'm not familiar with the differences between the series in that much detail. But it sounds reasonable to duplicate a bit of the PLL code in order to make sure it matches more exactly to a particular series.